api/envoy/api/v2/listener/udp_listener_config.proto

```diff
@@ -30,6 +30,17 @@ message UdpListenerConfig {
 
     google.protobuf.Any typed_config = 3;
   }
+  
+  enum UdpShutDownMode {
```

Comment:
You should probably have some `0` entry such as `UNSET`.

Reply:
@htuch was thinking about it, but it could lead to interesting state. What would be the logic when `udp_enabled` is false (we want to stop udp processing) and shutdown_mode is unset?

Reply:
I would probably just make the 0 entry the default which is graceful?

Reply:
Yeah, that is probably a reasonable default that won't change, let's do that.